### PR TITLE
Add support for latest libnx HID API

### PR DESCRIPTION
**Pull request by HamletDuFromage**

### CODE REVIEW

**Comment by fincs** [comment]
Good work, but this has nothing to do with AMS; it's related to libnx APIs.

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,7 @@ cmake_minimum_required(VERSION 3.0)
 project(GLFW VERSION 3.3.2 LANGUAGES C)
 
 set(CMAKE_LEGACY_CYGWIN_WIN32 OFF)
+set(SWITCH_LIBNX ON)
```
Comment (fincs) [inline CMakeLists.txt]: This isn't needed because it's set by the pkgbuild.

File: src/switch_joystick.c [hunk 2]
```diff
@@ -107,24 +107,29 @@ static _GLFWmapping s_switchMapping =
...
 {
     _GLFWjoystick* js = _glfwAllocJoystick(s_switchMapping.name, s_switchMapping.guid,
         _SWITCH_AXIS_COUNT, _SWITCH_BUTTON_COUNT, _SWITCH_HAT_COUNT);
 
+    padConfigureInput(1, HidNpadStyleSet_NpadStandard);
+    padInitializeDefault(&pad);
+
     js->mapping = &s_switchMapping;
-    js->nx.id = CONTROLLER_P1_AUTO;
+    padIsNpadActive (&pad, js->nx.id);
```
Comment (fincs) [inline src/switch_joystick.c]: This does nothing. Might want to remove this line, together with the "id" member (which doesn't really have any purpose anymore).

File: src/switch_joystick.c [hunk 3]
```diff
@@ -212,9 +219,3 @@ int _glfwPlatformPollJoystick(_GLFWjoystick* js, int mode)
     return GLFW_TRUE;
 }
 
-/* Not used on Switch
```
Comment (fincs) [inline src/switch_joystick.c]: Restore this, for documentation's sake.

File: src/switch_joystick.h [hunk 4]
```diff
@@ -33,7 +33,7 @@
 //
 typedef struct _GLFWjoystickNX
 {
-    HidControllerID id;
+    HidNpadIdType id;
```
Comment (fincs) [inline src/switch_joystick.h]: See comment above, I think we can remove this field.

File: CMakeLists.txt (hunk 1 above)
Comment (HamletDuFromage) [reply CMakeLists.txt]: makes sense, somehow I could compile it without it and forgot to remove it

File: src/switch_joystick.c [hunk 5]
```diff
@@ -107,24 +107,28 @@ static _GLFWmapping s_switchMapping =
...
 
+PadState pad;
+
 void _glfwInitSwitchJoysticks(void)
 {
     _GLFWjoystick* js = _glfwAllocJoystick(s_switchMapping.name, s_switchMapping.guid,
         _SWITCH_AXIS_COUNT, _SWITCH_BUTTON_COUNT, _SWITCH_HAT_COUNT);
 
+    padConfigureInput(1, HidNpadStyleSet_NpadStandard);
+    padInitializeDefault(&pad);
```
Comment (fincs) [inline src/switch_joystick.c:118]: Add a `hidInitializeTouchScreen();` call after this (forgot to mention it earlier)

File: src/switch_joystick.c [hunk 6]
```diff
@@ -216,5 +222,4 @@ int _glfwPlatformPollJoystick(_GLFWjoystick* js, int mode)
 void _glfwPlatformUpdateGamepadGUID(char* guid)
 {
 }
-*/
```
Comment (fincs) [inline src/switch_joystick.c]: Revert whitespace change (restore trailing newline)

**Comment by HamletDuFromage** [comment]
> Good work, but this has nothing to do with AMS; it's related to libnx APIs.

I knew it too, but got it confused for no reason :^)

Hopefully it's all good now

**Pull request merged**